docs/docs/reference/syntax.md

```diff
@@ -374,14 +374,12 @@ DefSig            ::=  id [DefTypeParamClause] DefParamClauses
 TypeDcl           ::=  id [TypeParamClause] {FunParamClause} TypeBounds [‘=’ Type]
 
 Def               ::=  ‘val’ PatDef
-                    |  ‘var’ VarDef
+                    |  ‘var’ PatDef
                     |  ‘def’ DefDef
                     |  ‘type’ {nl} TypeDcl
                     |  TmplDef
 PatDef            ::=  ids [‘:’ Type] ‘=’ Expr
                     |  Pattern2 [‘:’ Type] ‘=’ Expr
-VarDef            ::=  PatDef
-                    |  ids ‘:’ Type ‘=’ ‘_’
```

Comment:
We could avoid all the issues with inconsistent semantics and bad error messages if we define this as syntax.

```
VarDef            ::=  PatDef
                    |  ids ‘:’ Type ‘=’ ‘_’   // compat
                    |  ids ‘:’ Type ‘=’ ‘unimplemented’ 
```

Then simply desugar `var x: T = uninitialized ` into `var x: T = _` or any other internal representation we want. We could also desugar both into `var x: T = scala.runtime. uninitialized`. This would mean that all the checks and error messages for `var x: T = _` would also be used `var x: T = uninitialized`.

We could escape the syntax using ``var x: T = `uninitialized` `` if some app does use a `def uninitialized` already.

Reply:
This also has an advantage for the IDE and GitHub as we could highlight this `uninitialized` token.

Reply:
With this approach, we could still evolve the feature to somthing more complex in the future such as requiring an import and defining a method for it in the library. Those are the details that we should not rush to implement right now. It is only the syntax change that we care about.

Reply:
> Then simply desugar var x: T = uninitialized into var x: T = _ or any other internal representation we want.

That's actually what we do, in `PruneErasedDefs`. It's just that the official syntax does not mention `_ =` (just as it does not mention all the other 2.13 things we still support for now).

Reply:
No, I think that would be worse. We want a regular identifier that we can hover over in an IDE. I think @compileTimeOnly works very well for that. Otherwise you get into problems with soft keywords. It would be positively weird to _require_ a backticked `uninitialized` on the right hand side. That's just another obscure special case. We want to get away from these!